api/envoy/config/route/v3/route_components.proto

```diff
@@ -453,6 +453,20 @@ message WeightedCluster {
   }
 }
 
+// Configuration for a cluster specifier plugin.
+message ClusterSpecifierPlugin {
+  // The name of the plugin and its opaque configuration.
+  core.v3.TypedExtensionConfig extension = 1 [(validate.rules).message = {required: true}];
+
+  // [#not-implemented-hide:]
```

Comment:
If you're implementing this mechanism in Envoy, please also implement the `is_optional` field.